fission-web-server/library/Fission/Web/Server/Internal/Tasks.hs

```diff
@@ -45,7 +54,71 @@ deleteByUsername userNameTxt =
         Nothing                -> error "User doesn't exist"
 
 ---
+syncDNS :: Server()
+syncDNS = do
+  users <- getUserData
+  apps <- getAppData
+  mapM_ setUserDNS users
+  mapM_ setAppDNS apps
+
+getUserData :: Server [Entity User]
+getUserData = do
+  runDB do
+    select $ from \user -> do
+      where_ $ SQL.isNothing (user ^. UserSecretDigest)
+      return user
 
+getAppData :: Server [(Entity App, Entity AppDomain)]
+getAppData = do
+  runDB do
+    select $ from \(app `InnerJoin` appDomain) -> do
+      SQL.on $ appDomain ^. AppDomainAppId ==. app ^. AppId
+      return (app, appDomain)
+
+setUserDNS :: Entity User -> Server()
+setUserDNS (Entity userId User { userUsername, userPublicKey, userDataRoot }) = do
+  zoneID <- asks userZoneID
+  domainName <- asks userRootDomain
+  let
+    unSubDom = Subdomain $ textDisplay userUsername
+    url      = URL {domainName, subdomain = Just (Subdomain "_did") <> Just unSubDom}
+    segments = case userPublicKey of
+      Nothing   -> pure ""
+      Just pkey -> DNS.splitRecord $ textDisplay (DID Key pkey)
```

Comment:
This is SUCH a nit, but I've seen this in a few places in this PR and it makes my eye twitch 😛 

If you could put a blank line between an indented and the next outdented line, it's pretty idiomatic